src/hooks/useDeleteSavedSearch.tsx

```diff
@@ -0,0 +1,47 @@
+import React, {useState} from 'react';
+import ConfirmModal from '@components/ConfirmModal';
+import Navigation from '@libs/Navigation/Navigation';
+import * as SearchUtils from '@libs/SearchUtils';
+import * as SearchActions from '@userActions/Search';
+import CONST from '@src/CONST';
+import ROUTES from '@src/ROUTES';
+import useLocalize from './useLocalize';
+
+export default function useDeleteSavedSearch() {
+    const [isDeleteModalVisible, setIsDeleteModalVisible] = useState(false);
+    const [hashToDelete, setHashToDelete] = useState(0);
+    const {translate} = useLocalize();
+
+    const showDeleteModal = (hash: number) => {
+        setIsDeleteModalVisible(true);
+        setHashToDelete(hash);
+    };
+
+    const handleDelete = () => {
+        SearchActions.deleteSavedSearch(hashToDelete);
+        setIsDeleteModalVisible(false);
+        SearchActions.clearAdvancedFilters();
+        Navigation.navigate(
+            ROUTES.SEARCH_CENTRAL_PANE.getRoute({
+                query: SearchUtils.buildCannedSearchQuery(CONST.SEARCH.DATA_TYPES.EXPENSE, CONST.SEARCH.STATUS.EXPENSE.ALL),
+            }),
```

Comment:
Coming from https://github.com/Expensify/App/issues/57749, we should have cleared this only if the current hash was equal to the hash to delete.